plugin/trino-mysql/src/main/java/io/trino/plugin/mysql/MySqlClient.java

```diff
@@ -456,6 +471,62 @@ public Optional<ColumnMapping> toColumnMapping(ConnectorSession session, Connect
         return Optional.empty();
     }
 
+    private LongWriteFunction mySqlDateWriteFunctionUsingLocalDate()
+    {
+        LocalDate minMySqlDate = LocalDate.of(0, 1, 1);
+        return LongWriteFunction.of(Types.DATE, (statement, index, epochDay) -> {
+            LocalDate localDate = LocalDate.ofEpochDay(epochDay);
+            if (localDate.isBefore(minMySqlDate)) {
+                throw new TrinoException(INVALID_ARGUMENTS, format("Value must be after %s in MySQL: '%s'", minMySqlDate, localDate));
+            }
+            statement.setObject(index, localDate);
```

Comment:
If we use `PreparedStatement.setString` with bind expression, there's no need to validate in connector side?

Reply:
Good idea, let me try that.